### ISSUE TITLE
ClassCastException when deserializing JSONArray to Object

### ISSUE THREAD

**Issue by Maxr1998** [afollestad/ason#10]
When you have e.g. a class
```java
public class Example {
    public Object item;
}
```
and the data `Ason ason = new Ason("{\"id\": [1, 2, 3, 4]}")`, when you use `Ason.deserialize(ason, Example.class)` on it, it throws

```
java.lang.ClassCastException: com.afollestad.ason.AsonArray cannot be cast to com.afollestad.ason.Ason
    at com.afollestad.ason.AsonSerializer.deserialize(AsonSerializer.java:168)
    at com.afollestad.ason.Ason.deserialize(Ason.java:374)
    …
```

***
The error is probably at [AsonSerializer/Line 168](https://github.com/afollestad/ason/blob/master/src/main/java/com/afollestad/ason/AsonSerializer.java#L168), as it just assumes `Ason` as type when it can also be `AsonArray`.
```java
} else {
    Ason asonObject = ason.get(name);
    setFieldValue(field, newObject, deserialize(asonObject, type));
}
```
This could be a possible fix:
```java
} else {
    Object result = ason.get(name);
    Object value = null;
    if (result instanceof Ason) {
        value = deserialize((Ason) result, type);
    } else if (result instanceof AsonArray) {
        value = deserializeArray((AsonArray) result, type);
    }
    setFieldValue(field, newObject, value);
}
```

**Comment by afollestad** [comment]
@Maxr1998 this is now resolved and available in version 1.4.1.

Btw, "item" doesn't match up with "id" above 😀

**Comment by Maxr1998** [comment]
@afollestad thanks! I guess that happens when you write example code without checking it first at 11pm ;)
